include/ddc/discrete_domain.hpp

```diff
@@ -365,6 +365,24 @@ using convert_type_seq_to_discrete_domain = typename ConvertTypeSeqToDiscreteDom
 
 } // namespace detail
 
+template <typename... DDom>
+struct CartesianProd;
+
+template <typename... DDim1, typename... DDim2>
+struct CartesianProd<ddc::DiscreteDomain<DDim1...>, ddc::DiscreteDomain<DDim2...>>
+{
+    using type = ddc::DiscreteDomain<DDim1..., DDim2...>;
+};
+
+template <typename DDom1, typename DDom2, typename... Tail>
+struct CartesianProd<DDom1, DDom2, Tail...>
+{
+    using type = typename CartesianProd<typename CartesianProd<DDom1, DDom2>::type, Tail...>::type;
+};
+
+template <typename... DDom>
+using cartesian_prod = typename CartesianProd<DDom...>::type;
```

Comment:
same about `_t`

Reply:
in what case do we need this?

Reply:
In Gysela
using FdistribuDDom = ddc::cartesian_prod<SpDDom, SpatialDDom, VelocityDDom>;

Reply:
sounds nice. you can add the handling of the degenerate case `ddc::cartesian_prod<SingleDom>` while you're at it